edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/commands/KubernetesCrdCommandTest.cs

```diff
@@ -114,6 +114,66 @@ public async void CrdCommandExecuteWithAuthCreateNewObjects()
             }
         }
 
+        [Fact]
+        [Unit]
+        public async void CrdCommandExecuteDeploysModulesWithEnvVars()
+        {
+            CombinedDockerConfig config = new CombinedDockerConfig("image", new Docker.Models.CreateContainerParameters(), Option.None<AuthConfig>());
+            IDictionary<string, EnvVal> moduleEnvVars = new Dictionary<string, EnvVal>()
+            {
+                { "ACamelCaseEnvVar", new EnvVal("ACamelCaseEnvVarValue")}
+            };
+
+            IModule m1 = new DockerModule("module1", "v1", ModuleStatus.Running, Core.RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, moduleEnvVars);
+            var km1 = new KubernetesModule((IModule<DockerConfig>)m1, config);
+            KubernetesModule[] modules = { km1 };
+            var token = default(CancellationToken);
+            var auth = new AuthConfig() { Username = "username", Password = "password", ServerAddress = "docker.io" };
+            var configProvider = new Mock<ICombinedConfigProvider<CombinedDockerConfig>>();
+            configProvider.Setup(cp => cp.GetCombinedConfig(km1, Runtime)).Returns(() => new CombinedDockerConfig("test-image:1", Config1.CreateOptions, Option.Maybe(auth)));
+            EdgeDeploymentDefinition postedEdgeDeploymentDefinition = null;
+            bool postCrdCalled = false;
+
+            using (var server = new KubernetesApiServer(
+                resp: string.Empty,
+                shouldNext: httpContext =>
+                {
+                    string pathStr = httpContext.Request.Path.Value;
+                    string method = httpContext.Request.Method;
+                    if (string.Equals(method, "GET", StringComparison.OrdinalIgnoreCase))
+                    {
+                        httpContext.Response.StatusCode = 404;
+                    }
+                    else if (string.Equals(method, "POST", StringComparison.OrdinalIgnoreCase))
+                    {
+                        httpContext.Response.StatusCode = 201;
+                        httpContext.Response.Body = httpContext.Request.Body;
+                        if (pathStr.Contains($"namespaces/{Namespace}/{Constants.EdgeDeployment.Plural}"))
+                        {
+                            postCrdCalled = true;
+                            postedEdgeDeploymentDefinition = JsonConvert.DeserializeObject<EdgeDeploymentDefinition>(new StreamReader(httpContext.Response.Body)
+                                                                                                                         .ReadToEnd());
```

Comment:
nit: Could you please extract body content into a separate variable? It will make code more readable